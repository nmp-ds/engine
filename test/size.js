import { setup, getFractions, getGenerator } from './_helpers.js';
import { spaceBase } from '#theme';
import { describe, expect, test } from 'vitest';

setup();

const getSpecialSuffixes = (prefix) => ['full', 'fit', 'min', 'max', 'screen'].map(e => `${prefix}-${e}`);

describe('width and height', () => {
  test('width', async ({ uno }) => {
    const classes = spaceBase.map(e => `w-${e}`);
    classes.push(...getSpecialSuffixes('w'));
    classes.push(...getFractions('w'));
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test(`width without dash shouldn't match`, async ({ uno }) => {
    const { css } = await uno.generate(['w2', 'w32']);
    expect(css).toMatchInlineSnapshot('""');
  });
  test('height', async ({ uno }) => {
    const classes = spaceBase.map(e => `h-${e}`);
    classes.push(...getSpecialSuffixes('h'));
    classes.push(...getFractions('h'));
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test(`height without dash shouldn't match`, async ({ uno }) => {
    const { css } = await uno.generate(['h32']);
    expect(css).toMatchInlineSnapshot('""');
  });
});

describe('max width and height', () => {
  test('max-width', async ({ uno }) => {
    const classes = spaceBase.map(e => `max-w-${e}`);
    classes.push(...getSpecialSuffixes('max-w'));
    classes.push(...getFractions('max-w'));
    classes.push('max-w-prose');
    classes.push('max-w-none');
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-height', async ({ uno }) => {
    const classes = spaceBase.map(e => `max-h-${e}`);
    classes.push(...getSpecialSuffixes('max-h'));
    classes.push(...getFractions('max-h'));
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
});

describe('min width and height', () => {
  test('min-width', async ({ uno }) => {
    const classes = spaceBase.map(e => `min-w-${e}`);
    classes.push(...getSpecialSuffixes('min-w'));
    classes.push(...getFractions('min-w'));
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-height', async ({ uno }) => {
    const classes = spaceBase.map(e => `min-h-${e}`);
    classes.push(...getSpecialSuffixes('min-h'));
    classes.push(...getFractions('min-h'));
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
});


describe("width and height with arbitrary values", () => {
  test('width with css variables and anything else', async(t) => {
    const classes = ['w-[--css-variable]', 'w-[var(--css-variable)]', 'w-[customStuff]'];
    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });

  test('width with rem values', async (t) => {
    const classes = ['w-[20000]', 'w-[99999]', 'w-[378]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('width with rem unit', async (t) => {
    const classes = ['w-[20000rem]', 'w-[99999rem]', 'w-[378rem]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('width with pixel values', async () => {
    const classes = ['w-[20000]', 'w-[99999]', 'w-[378]'];

    const uno = getGenerator({ usePixels: true });
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('width with pixel unit', async (t) => {
    const classes = ['w-[20000px]', 'w-[99999px]', 'w-[378px]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('width with invalid value', async (t) => {
    const classes = ['w-20000', 'w-99999', 'w-378'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchInlineSnapshot('""');
  });
  test('height with css variables and anything else', async(t) => {
    const classes = ['h-[--css-variable]', 'h-[var(--css-variable)]', 'h-[customStuff]'];
    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('height with rem values', async (t) => {
    const classes = ['h-[20000]', 'h-[99999]', 'h-[378]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('height with rem unit', async (t) => {
    const classes = ['h-[20000rem]', 'h-[99999rem]', 'h-[378rem]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('height with pixel values', async () => {
    const classes = ['h-[20000]', 'h-[99999]', 'h-[378]'];

    const uno = getGenerator({ usePixels: true });
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('height with pixel unit', async (t) => {
    const classes = ['h-[20000px]', 'h-[99999px]', 'h-[378px]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('height with invalid value', async (t) => {
    const classes = ['h-20000', 'h-99999', 'h-378'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchInlineSnapshot('""');
  });
});

describe("min width and height with arbitrary values", () => {
  test('min width with css variables and anything else', async(t) => {
    const classes = ['min-w-[--css-variable]', 'min-w-[var(--css-variable)]', 'min-w-[customStuff]'];
    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-width with rem values', async (t) => {
    const classes = ['min-w-[20000]', 'min-w-[99999]', 'min-w-[378]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-width with rem unit', async (t) => {
    const classes = ['min-w-[20000rem]', 'min-w-[99999rem]', 'min-w-[378rem]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-width with pixel values', async () => {
    const classes = ['min-w-[20000]', 'min-w-[99999]', 'min-w-[378]'];

    const uno = getGenerator({ usePixels: true });
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-width with pixel unit', async (t) => {
    const classes = ['min-w-[20000px]', 'min-w-[99999px]', 'min-w-[378px]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-width with invalid value', async (t) => {
    const classes = ['min-w-20000', 'min-w-99999', 'min-w-378'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchInlineSnapshot('""');
  });

  test('min height with css variables and anything else', async(t) => {
    const classes = ['min-h-[--css-variable]', 'min-h-[var(--css-variable)]', 'min-h-[customStuff]'];
    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-height with rem values', async (t) => {
    const classes = ['min-h-[20000]', 'min-h-[99999]', 'min-h-[378]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-height with rem unit', async (t) => {
    const classes = ['min-h-[20000rem]', 'min-h-[99999rem]', 'min-h-[378rem]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-height with pixel values', async () => {
    const classes = ['min-h-[20000]', 'min-h-[99999]', 'min-h-[378]'];

    const uno = getGenerator({ usePixels: true });
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-height with pixel unit', async (t) => {
    const classes = ['min-h-[20000px]', 'min-h-[99999px]', 'min-h-[378px]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('min-height with invalid value', async (t) => {
    const classes = ['min-h-20000', 'min-h-99999', 'min-h-378'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchInlineSnapshot('""');
  });
});

describe("max width and height with arbitrary values", () => {
  test('max-width with rem values', async (t) => {
    const classes = ['max-w-[20000]', 'max-w-[99999]', 'max-w-[378]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-width with rem unit', async (t) => {
    const classes = ['max-w-[20000rem]', 'max-w-[99999rem]', 'max-w-[378rem]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-width with pixel values', async () => {
    const classes = ['max-w-[20000]', 'max-w-[99999]', 'max-w-[378]'];

    const uno = getGenerator({ usePixels: true });
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-width with pixel unit', async (t) => {
    const classes = ['max-w-[20000px]', 'max-w-[99999px]', 'max-w-[378px]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-width with invalid value', async (t) => {
    const classes = ['max-w-20000', 'max-w-99999', 'max-w-378'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchInlineSnapshot('""');
  });

  test('max-height with rem values', async (t) => {
    const classes = ['max-h-[20000]', 'max-h-[99999]', 'max-h-[378]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-height with rem unit', async (t) => {
    const classes = ['max-h-[20000rem]', 'max-h-[99999rem]', 'max-h-[378rem]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-height with pixel values', async () => {
    const classes = ['max-h-[20000]', 'max-h-[99999]', 'max-h-[378]'];

    const uno = getGenerator({ usePixels: true });
    const { css } = await uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-height with pixel unit', async (t) => {
    const classes = ['max-h-[20000px]', 'max-h-[99999px]', 'max-h-[378px]'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchSnapshot();
  });
  test('max-height with invalid value', async (t) => {
    const classes = ['max-h-20000', 'max-h-99999', 'max-h-378'];

    const { css } = await t.uno.generate(classes);
    expect(css).toMatchInlineSnapshot('""');
  });
});
