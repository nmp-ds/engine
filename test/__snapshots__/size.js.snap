// Vitest Snapshot v1

exports[`max width and height > max-height 1`] = `
"/* layer: default */
.max-h-0{max-height:0rem;}
.max-h-1{max-height:0.1rem;}
.max-h-1\\\\/1,
.max-h-2\\\\/2,
.max-h-3\\\\/3,
.max-h-4\\\\/4,
.max-h-5\\\\/5,
.max-h-6\\\\/6,
.max-h-full{max-height:100%;}
.max-h-1\\\\/2,
.max-h-2\\\\/4,
.max-h-3\\\\/6{max-height:50%;}
.max-h-1\\\\/3,
.max-h-2\\\\/6{max-height:33.3333333333%;}
.max-h-1\\\\/4{max-height:25%;}
.max-h-1\\\\/5{max-height:20%;}
.max-h-1\\\\/6{max-height:16.6666666667%;}
.max-h-10{max-height:1rem;}
.max-h-112{max-height:11.2rem;}
.max-h-12{max-height:1.2rem;}
.max-h-128{max-height:12.8rem;}
.max-h-14{max-height:1.4rem;}
.max-h-144{max-height:14.4rem;}
.max-h-16{max-height:1.6rem;}
.max-h-2{max-height:0.2rem;}
.max-h-2\\\\/3,
.max-h-4\\\\/6{max-height:66.6666666667%;}
.max-h-2\\\\/5{max-height:40%;}
.max-h-20{max-height:2rem;}
.max-h-24{max-height:2.4rem;}
.max-h-28{max-height:2.8rem;}
.max-h-3\\\\/4{max-height:75%;}
.max-h-3\\\\/5{max-height:60%;}
.max-h-32{max-height:3.2rem;}
.max-h-4{max-height:0.4rem;}
.max-h-4\\\\/5{max-height:80%;}
.max-h-40{max-height:4rem;}
.max-h-44{max-height:4.4rem;}
.max-h-48{max-height:4.8rem;}
.max-h-5\\\\/6{max-height:83.3333333333%;}
.max-h-56{max-height:5.6rem;}
.max-h-6{max-height:0.6rem;}
.max-h-64{max-height:6.4rem;}
.max-h-8{max-height:0.8rem;}
.max-h-80{max-height:8rem;}
.max-h-96{max-height:9.6rem;}
.max-h-fit{max-height:fit-content;}
.max-h-max{max-height:max-content;}
.max-h-min{max-height:min-content;}
.max-h-screen{max-height:100vh;}"
`;

exports[`max width and height > max-width 1`] = `
"/* layer: default */
.max-w-0{max-width:0rem;}
.max-w-1{max-width:0.1rem;}
.max-w-1\\\\/1,
.max-w-2\\\\/2,
.max-w-3\\\\/3,
.max-w-4\\\\/4,
.max-w-5\\\\/5,
.max-w-6\\\\/6,
.max-w-full{max-width:100%;}
.max-w-1\\\\/2,
.max-w-2\\\\/4,
.max-w-3\\\\/6{max-width:50%;}
.max-w-1\\\\/3,
.max-w-2\\\\/6{max-width:33.3333333333%;}
.max-w-1\\\\/4{max-width:25%;}
.max-w-1\\\\/5{max-width:20%;}
.max-w-1\\\\/6{max-width:16.6666666667%;}
.max-w-10{max-width:1rem;}
.max-w-112{max-width:11.2rem;}
.max-w-12{max-width:1.2rem;}
.max-w-128{max-width:12.8rem;}
.max-w-14{max-width:1.4rem;}
.max-w-144{max-width:14.4rem;}
.max-w-16{max-width:1.6rem;}
.max-w-2{max-width:0.2rem;}
.max-w-2\\\\/3,
.max-w-4\\\\/6{max-width:66.6666666667%;}
.max-w-2\\\\/5{max-width:40%;}
.max-w-20{max-width:2rem;}
.max-w-24{max-width:2.4rem;}
.max-w-28{max-width:2.8rem;}
.max-w-3\\\\/4{max-width:75%;}
.max-w-3\\\\/5{max-width:60%;}
.max-w-32{max-width:3.2rem;}
.max-w-4{max-width:0.4rem;}
.max-w-4\\\\/5{max-width:80%;}
.max-w-40{max-width:4rem;}
.max-w-44{max-width:4.4rem;}
.max-w-48{max-width:4.8rem;}
.max-w-5\\\\/6{max-width:83.3333333333%;}
.max-w-56{max-width:5.6rem;}
.max-w-6{max-width:0.6rem;}
.max-w-64{max-width:6.4rem;}
.max-w-8{max-width:0.8rem;}
.max-w-80{max-width:8rem;}
.max-w-96{max-width:9.6rem;}
.max-w-fit{max-width:fit-content;}
.max-w-max{max-width:max-content;}
.max-w-min{max-width:min-content;}
.max-w-none{max-width:none;}
.max-w-prose{max-width:65ch;}
.max-w-screen{max-width:100vw;}"
`;

exports[`min width and height > min-height 1`] = `
"/* layer: default */
.min-h-0{min-height:0rem;}
.min-h-1{min-height:0.1rem;}
.min-h-1\\\\/1,
.min-h-2\\\\/2,
.min-h-3\\\\/3,
.min-h-4\\\\/4,
.min-h-5\\\\/5,
.min-h-6\\\\/6,
.min-h-full{min-height:100%;}
.min-h-1\\\\/2,
.min-h-2\\\\/4,
.min-h-3\\\\/6{min-height:50%;}
.min-h-1\\\\/3,
.min-h-2\\\\/6{min-height:33.3333333333%;}
.min-h-1\\\\/4{min-height:25%;}
.min-h-1\\\\/5{min-height:20%;}
.min-h-1\\\\/6{min-height:16.6666666667%;}
.min-h-10{min-height:1rem;}
.min-h-112{min-height:11.2rem;}
.min-h-12{min-height:1.2rem;}
.min-h-128{min-height:12.8rem;}
.min-h-14{min-height:1.4rem;}
.min-h-144{min-height:14.4rem;}
.min-h-16{min-height:1.6rem;}
.min-h-2{min-height:0.2rem;}
.min-h-2\\\\/3,
.min-h-4\\\\/6{min-height:66.6666666667%;}
.min-h-2\\\\/5{min-height:40%;}
.min-h-20{min-height:2rem;}
.min-h-24{min-height:2.4rem;}
.min-h-28{min-height:2.8rem;}
.min-h-3\\\\/4{min-height:75%;}
.min-h-3\\\\/5{min-height:60%;}
.min-h-32{min-height:3.2rem;}
.min-h-4{min-height:0.4rem;}
.min-h-4\\\\/5{min-height:80%;}
.min-h-40{min-height:4rem;}
.min-h-44{min-height:4.4rem;}
.min-h-48{min-height:4.8rem;}
.min-h-5\\\\/6{min-height:83.3333333333%;}
.min-h-56{min-height:5.6rem;}
.min-h-6{min-height:0.6rem;}
.min-h-64{min-height:6.4rem;}
.min-h-8{min-height:0.8rem;}
.min-h-80{min-height:8rem;}
.min-h-96{min-height:9.6rem;}
.min-h-fit{min-height:fit-content;}
.min-h-max{min-height:max-content;}
.min-h-min{min-height:min-content;}
.min-h-screen{min-height:100vh;}"
`;

exports[`min width and height > min-width 1`] = `
"/* layer: default */
.min-w-0{min-width:0rem;}
.min-w-1{min-width:0.1rem;}
.min-w-1\\\\/1,
.min-w-2\\\\/2,
.min-w-3\\\\/3,
.min-w-4\\\\/4,
.min-w-5\\\\/5,
.min-w-6\\\\/6,
.min-w-full{min-width:100%;}
.min-w-1\\\\/2,
.min-w-2\\\\/4,
.min-w-3\\\\/6{min-width:50%;}
.min-w-1\\\\/3,
.min-w-2\\\\/6{min-width:33.3333333333%;}
.min-w-1\\\\/4{min-width:25%;}
.min-w-1\\\\/5{min-width:20%;}
.min-w-1\\\\/6{min-width:16.6666666667%;}
.min-w-10{min-width:1rem;}
.min-w-112{min-width:11.2rem;}
.min-w-12{min-width:1.2rem;}
.min-w-128{min-width:12.8rem;}
.min-w-14{min-width:1.4rem;}
.min-w-144{min-width:14.4rem;}
.min-w-16{min-width:1.6rem;}
.min-w-2{min-width:0.2rem;}
.min-w-2\\\\/3,
.min-w-4\\\\/6{min-width:66.6666666667%;}
.min-w-2\\\\/5{min-width:40%;}
.min-w-20{min-width:2rem;}
.min-w-24{min-width:2.4rem;}
.min-w-28{min-width:2.8rem;}
.min-w-3\\\\/4{min-width:75%;}
.min-w-3\\\\/5{min-width:60%;}
.min-w-32{min-width:3.2rem;}
.min-w-4{min-width:0.4rem;}
.min-w-4\\\\/5{min-width:80%;}
.min-w-40{min-width:4rem;}
.min-w-44{min-width:4.4rem;}
.min-w-48{min-width:4.8rem;}
.min-w-5\\\\/6{min-width:83.3333333333%;}
.min-w-56{min-width:5.6rem;}
.min-w-6{min-width:0.6rem;}
.min-w-64{min-width:6.4rem;}
.min-w-8{min-width:0.8rem;}
.min-w-80{min-width:8rem;}
.min-w-96{min-width:9.6rem;}
.min-w-fit{min-width:fit-content;}
.min-w-max{min-width:max-content;}
.min-w-min{min-width:min-content;}
.min-w-screen{min-width:100vw;}"
`;

exports[`width and height > height 1`] = `
"/* layer: default */
.h-0{height:0rem;}
.h-1{height:0.1rem;}
.h-1\\\\/1,
.h-2\\\\/2,
.h-3\\\\/3,
.h-4\\\\/4,
.h-5\\\\/5,
.h-6\\\\/6,
.h-full{height:100%;}
.h-1\\\\/2,
.h-2\\\\/4,
.h-3\\\\/6{height:50%;}
.h-1\\\\/3,
.h-2\\\\/6{height:33.3333333333%;}
.h-1\\\\/4{height:25%;}
.h-1\\\\/5{height:20%;}
.h-1\\\\/6{height:16.6666666667%;}
.h-10{height:1rem;}
.h-112{height:11.2rem;}
.h-12{height:1.2rem;}
.h-128{height:12.8rem;}
.h-14{height:1.4rem;}
.h-144{height:14.4rem;}
.h-16{height:1.6rem;}
.h-2{height:0.2rem;}
.h-2\\\\/3,
.h-4\\\\/6{height:66.6666666667%;}
.h-2\\\\/5{height:40%;}
.h-20{height:2rem;}
.h-24{height:2.4rem;}
.h-28{height:2.8rem;}
.h-3\\\\/4{height:75%;}
.h-3\\\\/5{height:60%;}
.h-32{height:3.2rem;}
.h-4{height:0.4rem;}
.h-4\\\\/5{height:80%;}
.h-40{height:4rem;}
.h-44{height:4.4rem;}
.h-48{height:4.8rem;}
.h-5\\\\/6{height:83.3333333333%;}
.h-56{height:5.6rem;}
.h-6{height:0.6rem;}
.h-64{height:6.4rem;}
.h-8{height:0.8rem;}
.h-80{height:8rem;}
.h-96{height:9.6rem;}
.h-fit{height:fit-content;}
.h-max{height:max-content;}
.h-min{height:min-content;}
.h-screen{height:100vh;}"
`;

exports[`width and height > width 1`] = `
"/* layer: default */
.w-0{width:0rem;}
.w-1{width:0.1rem;}
.w-1\\\\/1,
.w-2\\\\/2,
.w-3\\\\/3,
.w-4\\\\/4,
.w-5\\\\/5,
.w-6\\\\/6,
.w-full{width:100%;}
.w-1\\\\/2,
.w-2\\\\/4,
.w-3\\\\/6{width:50%;}
.w-1\\\\/3,
.w-2\\\\/6{width:33.3333333333%;}
.w-1\\\\/4{width:25%;}
.w-1\\\\/5{width:20%;}
.w-1\\\\/6{width:16.6666666667%;}
.w-10{width:1rem;}
.w-112{width:11.2rem;}
.w-12{width:1.2rem;}
.w-128{width:12.8rem;}
.w-14{width:1.4rem;}
.w-144{width:14.4rem;}
.w-16{width:1.6rem;}
.w-2{width:0.2rem;}
.w-2\\\\/3,
.w-4\\\\/6{width:66.6666666667%;}
.w-2\\\\/5{width:40%;}
.w-20{width:2rem;}
.w-24{width:2.4rem;}
.w-28{width:2.8rem;}
.w-3\\\\/4{width:75%;}
.w-3\\\\/5{width:60%;}
.w-32{width:3.2rem;}
.w-4{width:0.4rem;}
.w-4\\\\/5{width:80%;}
.w-40{width:4rem;}
.w-44{width:4.4rem;}
.w-48{width:4.8rem;}
.w-5\\\\/6{width:83.3333333333%;}
.w-56{width:5.6rem;}
.w-6{width:0.6rem;}
.w-64{width:6.4rem;}
.w-8{width:0.8rem;}
.w-80{width:8rem;}
.w-96{width:9.6rem;}
.w-fit{width:fit-content;}
.w-max{width:max-content;}
.w-min{width:min-content;}
.w-screen{width:100vw;}"
`;
